pkg/clusteragent/autoscaling/workload/provider.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package workload
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection"
+)
+
+// StartWorkloadAutoscaling starts the workload autoscaling controller
+func StartWorkloadAutoscaling(ctx context.Context, apiCl *apiserver.APIClient, rcClient rcClient) error {
+	if apiCl == nil {
+		return fmt.Errorf("Impossible to start workload autoscaling without valid APIClient")
+	}
+
+	le, err := leaderelection.GetLeaderEngine()
+	if err != nil {
+		return fmt.Errorf("Unable to start workload autoscaling as LeaderElection failed with: %v", err)
+	}
+
+	store := autoscaling.NewStore[model.PodAutoscalerInternal]()
+
+	_, err = newConfigRetriever(store, le.IsLeader, rcClient)
+	if err != nil {
+		return fmt.Errorf("Unable to start workload autoscaling config retriever: %w", err)
+	}
+
+	controller, err := newController(apiCl.RESTMapper, apiCl.ScaleCl, apiCl.DynamicInformerCl, apiCl.DynamicInformerFactory, le.IsLeader, store)
```

Comment:
⚠️ We're using `apiCl.DynamicInformerFactory` but the factory is behind some config flags:
https://github.com/DataDog/datadog-agent/blob/baef9cffc6c6ae21cf966d47ecb0c9699748d141/pkg/util/kubernetes/apiserver/apiserver.go#L329-L336